warehouse/query-core/src/main/java/datawave/query/transformer/UniqueTransform.java

```diff
@@ -79,279 +69,50 @@ public void updateConfig(UniqueFields uniqueFields, QueryModel model) {
-            }
-            orderedFieldSets.add(orderedFieldSet);
-        }
-        Collections.sort(orderedFieldSets);
-        return orderedFieldSets;
+        return null;
     }
     
     /**
-     * This will return attributes from a document that uniquely identify this document for a set of fields. The attributes will be organized as set of
-     * attribute sets.
-     *
-     * Definitions using example of "field.a.b.x = y" fieldname: "field" grouping context: "a.x" (the first part of the grouping is the group, the last part is
-     * the instance) value: "y" The unique fields to be grouped are specified as a set of fieldnames.
-     *
-     * The attributes that uniquely identify this document will actually be composed of multiple sets of attributes where the grouping context is consistent
-     * within each set.
-     *
-     * Example: Document: field1.a.1.0 = 1 field2.a.2.0 = 2 field1.a.1.1 = 3 field3.c.3.0 = 10 field3 = 11 field3 = 12 field4 = 100 ... unique fields = field1,
-     * field2, field3, field4 Resulting groups: field1 = 1, field2 = 2, field3 = 10, field4 = 100 field1 = 1, field2 = 2, field3 = 11/12, field4 = 100 field1 =
-     * 3, field2 = N/A, field3 = 10, field4 = 100 field1 = 3, field2 = N/A, field3 = 11/12, field4 = 100
+     * Returns the next unique document in the pipeline, if there is one. Also checks to see if the timeout for the page has exceeded and interjects an
+     * intermediate result instead of hte next unique document if so.
      *
+     * @return The next unique document per the configured fields, or an intermediate result (blank document) if the page execution time has elapsed, or null if
+     *         there isn't one and the timeout hasn't elapsed.
      */
-    private Set<Multimap<String,String>> getFieldSets(Document document) {
-        Map<String,Multimap<String,String>> mapGroupingContextToField = new HashMap<>();
-        for (String documentField : document.getDictionary().keySet()) {
-            String field = getUniqueField(documentField);
-            if (field != null) {
-                String groupingContext = getGroupingContext(documentField);
-                Set<String> values = getValues(document.get(documentField));
-                Set<String> transformedValues = uniqueFields.transformValues(field, values);
-                Multimap<String,String> groupedValues = mapGroupingContextToField.get(groupingContext);
-                if (groupedValues == null) {
-                    groupedValues = HashMultimap.create();
-                    mapGroupingContextToField.put(groupingContext, groupedValues);
-                }
-                groupedValues.putAll(field, transformedValues);
-            }
+    @Override
+    public Entry<Key,Document> flush() {
+        long elapsedExecutionTimeForCurrentPage = System.currentTimeMillis() - this.queryExecutionForPageStartTime;
+        if (elapsedExecutionTimeForCurrentPage > this.queryExecutionForPageTimeout) {
```

Comment:
We should only ever return "intermediate results" if there are potentially more results to return.  How do we know if the underlying scanners are finished or not?  I am missing something here.

Reply:
Ignoring this comment in favor of our in person conversation, and subsequent changes made.